test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Converters/IconUrlToImageCacheConverterTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Threading;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test
+{
+    public class IconUrlToImageCacheConverterTests
+    {
+        private static readonly ImageSource DefaultPackageIcon;
+
+        static IconUrlToImageCacheConverterTests()
+        {
+            DefaultPackageIcon = BitmapSource.Create(1, 1, 96, 96, PixelFormats.Bgr24, null, new byte[3] { 0, 0, 0 }, 3);
+            DefaultPackageIcon.Freeze();
+        }
+
+        [Fact]
+        public void Convert_WithMalformedUrlScheme_ReturnsDefault()
+        {
+            var iconUrl = new Uri("ttp://fake.com/image.png");
+
+            var converter = new IconUrlToImageCacheConverter();
+
+            var image = converter.Convert(
+                iconUrl, 
+                typeof(ImageSource),
+                DefaultPackageIcon, 
+                Thread.CurrentThread.CurrentCulture);
+
+            Assert.Same(DefaultPackageIcon, image);
+        }
+
+        [Fact]
+        public void Convert_WithLocalPath_ReturnsDefault()
+        {
+            var iconUrl = new Uri(@"C:\path\to\image.png");
+
+            var converter = new IconUrlToImageCacheConverter();
+
+            var image = converter.Convert(
+                iconUrl,
+                typeof(ImageSource),
+                DefaultPackageIcon,
+                Thread.CurrentThread.CurrentCulture);
+
+            Assert.Same(DefaultPackageIcon, image);
+        }
+
+        [Fact]
+        public void Convert_WithValidImageUrl_DownloadsImage()
+        {
+            var iconUrl = new Uri("http://go.microsoft.com/fwlink/?LinkID=288859");
```

Comment:
add a test for an invalid image (so a valid http link leading to an html page such as - https://dist.nuget.org/index.html